src/Framework/MSBuildEventSource.cs

```diff
@@ -0,0 +1,338 @@
+﻿using System;
+using System.IO;
+using System.Diagnostics.Tracing;
+using System.Globalization;
+
+
+//
+// This captures information of how various key methods of building with MSBuild ran.
+//
+namespace Microsoft.Build.Eventing
+{
+    // /OnlyProviders=*MSBuild-Profiling
+    [EventSource(Name = "MSBuild-Profiling")]
+    public sealed class MSBuildEventSource : EventSource
+    {
+        #region Singleton instance
+
+        // define the singleton instance of the event source
+        public static MSBuildEventSource Log = new MSBuildEventSource();
+
+        private MSBuildEventSource() { }
+
+        #endregion
+
+        #region Events and NonEvents
+
+        /// <summary>
+        /// Call this method to notify listeners of information relevant to collecting a set of items, mutating them in a specified way, and saving the results.
+        /// </summary>
+        public void ApplyStart()
```

Comment:
Each event should also have an `EventAttribute` on it.  So for ApplyStart it would be `[Event(1)]` where the 1 matches the value that you pass to WriteEvent.